#### Problem Statement:
Create a machine learning model to predict revenue of the restaurant based on
the features present in the dataset.

-Prerak pandya

In [1]:
#First We Import Lib for data cleaning
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

In [2]:
df = pd.read_csv(r"C:\Users\Prerak\Desktop\Resurent Price Prediction\revenue_prediction.csv")

In [3]:
df

Id                          Name Franchise       Category       City  \
0   101            HungryHowie'sPizza       Yes        Mexican  Bengaluru   
1   102          CharleysPhillySteaks        No    Varied Menu   Gurugram   
2   103                        Chuy's       Yes        Chicken       Pune   
3   104                   O'Charley's       Yes  Italian/Pizza     Mumbai   
4   105                 PolloTropical       Yes          Pizza      Noida   
..  ...                           ...       ...            ...        ...   
95  196              Wetzel'sPretzels        No  Italian/Pizza  Bengaluru   
96  197  LaMadeleineCountryFrenchCafe       Yes    Varied Menu  Bengaluru   
97  198                    Giordano's       Yes    Varied Menu   Gurugram   
98  199     IslandsFineBurgers&Drinks       Yes     Sports Bar       Pune   
99  200           Mimi'sBistro&Bakery        No            BBQ     Mumbai   

    No_Of_Item  Order_Placed  Revenue  
0           55           5.5  5953753  
1           72           6.8  7223131  
2           25           1.9  2555379  
3           18           2.5  2175511  
4           48           4.2  4816715  
..         ...           ...      ...  
95          19           1.1  1270499  
96          75           6.3  6412623  
97          77           6.2  6694797  
98          25           2.1  2344689  
99          50           4.4  4567678  

[100 rows x 8 columns]

In [4]:
# Here we have only 100 rows for prediction 
# so first we understand our data 
# 1st we going to clean data for good accurancy

In [5]:
df.shape

(100, 8)

In [6]:
df.isnull().sum()
#  check null value using isnull function

Id              0
Name            0
Franchise       0
Category        0
City            0
No_Of_Item      0
Order_Placed    0
Revenue         0
dtype: int64

In [8]:
df.duplicated().sum()
# check duplicate value using duplicate function

0

In [10]:
df.describe()
# check stastastics number using describe for better understand the data

Id  No_Of_Item  Order_Placed       Revenue
count  100.000000  100.000000    100.000000  1.000000e+02
mean   150.500000   49.080000      4.086000  4.395161e+06
std     29.011492   22.370923      2.055101  2.659932e+06
min    101.000000   18.000000      1.000000  8.498700e+05
25%    125.750000   34.750000      2.750000  2.688328e+06
50%    150.500000   45.000000      3.650000  3.911401e+06
75%    175.250000   57.250000      5.100000  5.330084e+06
max    200.000000  126.000000     13.000000  1.969694e+07

In [11]:
df.dtypes
#check data type
# we have see there is no problem in data type

Id                int64
Name             object
Franchise        object
Category         object
City             object
No_Of_Item        int64
Order_Placed    float64
Revenue           int64
dtype: object

In [13]:
#first we check unique values in Category
df['Category'].unique()

array(['Mexican', 'Varied Menu', 'Chicken', 'Italian/Pizza', 'Pizza',
       'Seafood', 'Steak', 'Frozen Desserts', 'Coffee Cafe', 'BBQ',
       'Sandwich', 'Bakery Cafe', 'Burger', 'Asian', 'Family Style',
       'Sports Bar', 'Asian/Noodle', 'Family Casual', 'Snack', 'Healthy'],
      dtype=object)

So many values we have so we drop this column we drop object datatype columns

In [19]:
df = df.drop(columns= ['City','Franchise','Name'])

In [20]:
df

Id  No_Of_Item  Order_Placed  Revenue
0   101          55           5.5  5953753
1   102          72           6.8  7223131
2   103          25           1.9  2555379
3   104          18           2.5  2175511
4   105          48           4.2  4816715
..  ...         ...           ...      ...
95  196          19           1.1  1270499
96  197          75           6.3  6412623
97  198          77           6.2  6694797
98  199          25           2.1  2344689
99  200          50           4.4  4567678

[100 rows x 4 columns]

In [21]:
#now we have predict the data 
# our target column is Revenue and revanue column have cotinues data  so i use linerregression for that


In [78]:
#The first step is to choose independent and dependent variables.
x = df.iloc[:,:-1]
y = df[['Revenue']]
#x is independent column
#y is target column

In [79]:
#split Data 
from sklearn.model_selection import train_test_split

In [80]:
x_train,x_test,y_train,y_test = train_test_split(x,y,test_size=0.2,random_state=23)

In [81]:
#now import Lninerregression lib
from sklearn.linear_model import LinearRegression

In [82]:
li = LinearRegression()


In [83]:
model = li.fit(x_train,y_train)
#train out model

In [84]:
prediction = model.predict(x_test)
#predict our model

In [85]:
from sklearn.metrics import r2_score,mean_absolute_error,mean_squared_error
#check accurancy with help of r2_score

In [86]:
r2_score(prediction,y_test)

0.8342653250221544

###### our model is ready and 83% of our model is accurate